### ISSUE TITLE
Titlebar doesn't honor user settings outside git environment

### ISSUE THREAD

**Issue by ebelew** [dahlbyk/posh-git#95]
I have a custom prompt function. It sets the titlebar and the command line with special values:

``` powershell
function prompt {
    $realLASTEXITCODE = $LASTEXITCODE

    $cliPrefix   = "$(getCurrentUser) $($pwd.ProviderPath)"
    Write-Host -NoNewline $cliPrefix

    $titlePrefix = "$(getCurrentUser) $(Split-Path -Leaf $pwd.ProviderPath)"
    $Host.UI.RawUI.WindowTitle = $titlePrefix
    $Global:PreviousWindowTitle = $titlePrefix
    $Global:GitPromptSettings.EnableWindowTitle = $titlePrefix

    # Reset color, which can be messed up by Enable-GitColors
    $Host.UI.RawUI.ForegroundColor = $GitPromptSettings.DefaultForegroundColor

    Write-VcsStatus

    $global:LASTEXITCODE = $realLASTEXITCODE
    return "> "
}
```

If I don't set $Global:PreviousWindowTitle, then my custom title gets overwritten every time Write-VcsStatus is called. Hence, once I go into a git repo, then back out, the title isn't updated.

It seems silly that I have to set my title in 2 places in order for it to actually get used. 
Note: The first setting of Host.UI.RawUI.WindowTitle is currently superfluous, since Write-VcsStatus always overwrites it

In summary, here are the repro steps:
In prompt, set window title, then call Write-VcsStatus
Set-Location into a git repo
Set-Location outside a git repo
Titlebar still contains previous repo name, not personalized titlebar

**Comment by dahlbyk** [comment]
Is this still a problem?

**Comment by rkeithhill** [comment]
I tried the repro steps and couldn't repro it.  However, I don't like this related script:
```powershell
$WindowTitleSupported = $true
if (Get-Module NuGet) {
    $WindowTitleSupported = $false
}
```
I believe the intent here is to see if we are running under the VS NuGet Pkg Mgr console.  If so, the above test should be changed to `if ($Host.Name -eq 'Package Manager Host')`.  Or perhaps we invert the logic and test against host name `ConsoleHost`. That would probably be safer. Or, we could just set the WindowTitle all the time and wrap with a try/catch.

**Comment by dahlbyk** [comment]
@rkeithhill I'm going to go ahead and close this, but feel free to open an issue for discussing potential improvements to the window title situation. Note also the discussion over in https://github.com/dahlbyk/posh-git/pull/96.
